UE4SS/src/LuaType/LuaUEnum.cpp

```diff
@@ -110,16 +110,199 @@ No overload found for function 'UEnum.ForEachName'.
             return 0;
         });
 
-        if constexpr (is_final == LuaMadeSimple::Type::IsFinal::Yes)
-        {
-            table.add_pair("type", [](const LuaMadeSimple::Lua& lua) -> int {
-                lua.set_string(ClassName::ToString());
-                return 1;
-            });
-
-            // If this is the final object then we also want to finalize creating the table
-            // If not then it's the responsibility of the overriding object to call 'make_global()'
-            // table.make_global(ClassName::ToString());
-        }
+        table.add_pair("GetEnumNameByIndex", [](const LuaMadeSimple::Lua& lua) -> int {
+            std::string error_overload_not_found{R"(
+No overload found for function 'UEnum.InsertIntoNames'.
+Overloads:
+#1: GetNameByValue(integer Value))"};
+
+            auto& lua_object = lua.get_userdata<UEnum>();
+
+            if (!lua.is_integer())
+            {
+                lua.throw_error(error_overload_not_found);
+            }
+
+            auto value = lua.get_integer();
+            LuaType::FName::construct(lua, lua_object.get_remote_cpp_object()->GetEnumNameByIndex(value));
+            return 1;
+        });
+
+        table.add_pair("InsertIntoNames", [](const LuaMadeSimple::Lua& lua) -> int {
```

Comment:
This is not how `InsertIntoNames` works.
The first param is the FName/int64 pair, the second param (the 'index') is the index into the array where this new pair will be inserted.
In Lua, since we don't have a pair, we have a couple of choices:
1. Use an additional param, making the signature: `InsertIntoNames(string Name, integer Value, integer Index, bool bShiftValues)`.
2. Use a Lua table to represent the Name/Value pair.

I have not reviewed the rest of this function because it needs to be reworked.
Will review this function after the rework.

Reply:
Reworked this. Let me know if there's any issues.